src/main/scala/beam/utils/Statistics.scala

```diff
@@ -64,4 +64,19 @@ object Statistics {
       )
     }
   }
+
+  /***
+    * If there is no weights then just get a plain average  (v1 + v2 + ... + vN)/N,
+    * otherwise calculated weighted average (w1*v1 + w2*v2 + ... + wN*vN)/(w1 + w2 + ... + wN)
+    */
+  private def average(values: Seq[Double], weights: Seq[Double]): Double = {
+    if (weights.isEmpty) {
+      values.sum / values.length
+    } else {
+      val counter = values.view.zip(weights).foldLeft((0d, 0d)) {
```

Comment:
nitpick: I'd suggest to reuse `Mean` type from common-math to compute the weighted average: commons.apache.org/proper/commons-math/javadocs/api-3.4.1/org/apache/commons/math3/stat/descriptive/moment/Mean.html#evaluate(double[],%20double[])

Otherwise if you leave it as is, please, add some tests.

Reply:
I used Mean instead of custom implementation.